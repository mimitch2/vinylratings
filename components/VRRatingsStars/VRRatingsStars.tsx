import React from 'react';
import { View } from 'react-native';
import { useTheme } from '@react-navigation/native';

import { VRIcon } from 'components';
import { generateArrayOfNumbers, getRatingValues } from 'helpers';
import { Theme } from 'constants/index';
import { IconSize } from 'types';

const RatingStar = ({
    inputRating,
    average,
    size = 'md'
}: {
    inputRating: number;
    average: number;
    size: IconSize;
}) => {
    const { colors }: Theme = useTheme();
    const { ratingInteger, ratingFloat } = getRatingValues({
        average
    });
    const shouldRenderFullStar =
        ratingInteger >= inputRating ||
        (ratingInteger + 1 === inputRating && ratingFloat >= 75);

    const shouldRenderHalfStar =
        ratingFloat >= 25 &&
        ratingFloat <= 75 &&
        ratingInteger === inputRating - 1;

    const type = shouldRenderFullStar
        ? 'starFull'
        : shouldRenderHalfStar
        ? 'starHalf'
        : 'starEmpty';

    return (
        <VRIcon
            type={type}
            color={colors.primary}
            key={inputRating}
            size={size}
        />
    );
};

const VRRatingsStars = ({
    average,
    size = 'md'
}: {
    average: number;
    size?: IconSize;
}) => {
    return (
        <View style={{ flexDirection: 'row' }}>
            {generateArrayOfNumbers({ length: 5 }).map((inputRating) => {
                return (
                    <RatingStar
                        key={inputRating}
                        inputRating={inputRating}
                        average={average}
                        size={size}
                    />
                );
            })}
        </View>
    );
};

export default VRRatingsStars;
