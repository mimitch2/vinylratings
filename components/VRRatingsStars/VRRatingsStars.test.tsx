import React from 'react';

import VRRatingsStars from './VRRatingsStars';
import { renderWithProvider } from 'test';

describe('VRRatingsStars', () => {
    it('should render correctly for 0 stars', () => {
        const view = renderWithProvider(<VRRatingsStars average={0} />);

        expect(view).toMatchSnapshot();
    });

    it('should render correctly for 3.5 stars', () => {
        const view = renderWithProvider(<VRRatingsStars average={3.5} />);

        expect(view).toMatchSnapshot();
    });

    it('should render correctly for 5 stars', () => {
        const view = renderWithProvider(<VRRatingsStars average={3.5} />);

        expect(view).toMatchSnapshot();
    });

    it('should render different size', () => {
        const view = renderWithProvider(
            <VRRatingsStars average={3.5} size="sm" />
        );

        expect(view).toMatchSnapshot();
    });
});
