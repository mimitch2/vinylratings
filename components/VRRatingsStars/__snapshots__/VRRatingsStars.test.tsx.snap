// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VRRatingsStars should render correctly for 0 stars 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#222B45",
      },
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m16.15 37.75 7.85-4.7 7.85 4.75-2.1-8.9 6.9-6-9.1-.8L24 13.7l-3.55 8.35-9.1.8 6.9 6ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55ZM24 26.25Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m16.15 37.75 7.85-4.7 7.85 4.75-2.1-8.9 6.9-6-9.1-.8L24 13.7l-3.55 8.35-9.1.8 6.9 6ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55ZM24 26.25Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m16.15 37.75 7.85-4.7 7.85 4.75-2.1-8.9 6.9-6-9.1-.8L24 13.7l-3.55 8.35-9.1.8 6.9 6ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55ZM24 26.25Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m16.15 37.75 7.85-4.7 7.85 4.75-2.1-8.9 6.9-6-9.1-.8L24 13.7l-3.55 8.35-9.1.8 6.9 6ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55ZM24 26.25Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m16.15 37.75 7.85-4.7 7.85 4.75-2.1-8.9 6.9-6-9.1-.8L24 13.7l-3.55 8.35-9.1.8 6.9 6ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55ZM24 26.25Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`VRRatingsStars should render correctly for 3.5 stars 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#222B45",
      },
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M24 13.7v19.35l7.85 4.75-2.1-8.9 6.9-6-9.1-.8ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m16.15 37.75 7.85-4.7 7.85 4.75-2.1-8.9 6.9-6-9.1-.8L24 13.7l-3.55 8.35-9.1.8 6.9 6ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55ZM24 26.25Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`VRRatingsStars should render correctly for 5 stars 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#222B45",
      },
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M24 13.7v19.35l7.85 4.75-2.1-8.9 6.9-6-9.1-.8ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="25"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 25,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 25,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m16.15 37.75 7.85-4.7 7.85 4.75-2.1-8.9 6.9-6-9.1-.8L24 13.7l-3.55 8.35-9.1.8 6.9 6ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55ZM24 26.25Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`VRRatingsStars should render different size 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#222B45",
      },
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 20,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 20,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 20,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 20,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 20,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 20,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m11.65 44 3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 20,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 20,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M24 13.7v19.35l7.85 4.75-2.1-8.9 6.9-6-9.1-.8ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#222B45",
        },
        {},
      ]
    }
    testID="vr-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "height": 20,
            "marginHorizontal": -2,
            "marginTop": -2,
            "width": 20,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294743098,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m16.15 37.75 7.85-4.7 7.85 4.75-2.1-8.9 6.9-6-9.1-.8L24 13.7l-3.55 8.35-9.1.8 6.9 6ZM11.65 44l3.25-14.05L4 20.5l14.4-1.25L24 6l5.6 13.25L44 20.5l-10.9 9.45L36.35 44 24 36.55ZM24 26.25Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
